### ISSUE TITLE
networkctl --json=short reports: Failed to get description: Invalid argument

### ISSUE THREAD

**Issue by dave4445** [systemd/systemd#25089]
### systemd version the issue has been seen with

250.4

### Used distribution

buildroot 2022.02.x LTS

### Linux kernel version used

4.4.247

### CPU architectures issue was seen on

aarch64

### Component

networkctl

### Expected behaviour you didn't see

JSON formatted output

### Unexpected behaviour you saw

``` 
# networkctl list
IDX LINK TYPE     OPERATIONAL SETUP
  1 lo   loopback carrier     unmanaged
  2 eth0 ether    routable    configured

2 links listed.
# networkctl --json=short list
Failed to get description: Invalid argument
# 
```

### Steps to reproduce the problem

I'm trying to use `networkctl` to get JSON output, however the JSON feature doesn't work for some reason

### Additional program output to the terminal or log subsystem illustrating the issue

```sh
# SYSTEMD_LOG_LEVEL=debug networkctl --json=short list |cat
Bus n/a: changing state UNSET â OPENING
sd-bus: starting bus by connecting to /run/dbus/system_bus_socket...
Bus n/a: changing state OPENING â AUTHENTICATING
Bus n/a: changing state AUTHENTICATING â HELLO
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=Hello cookie=
1 reply_cookie=0 signature=n/a error-name=n/a error-message=n/a
Got message type=method_return sender=org.freedesktop.DBus destination=:1.38 path=n/a interface=n/a member=n/a cookie=1 reply_cookie=1 signature=s error
-name=n/a error-message=n/a
Bus n/a: changing state HELLO â RUNNING
Sent message type=method_call sender=n/a destination=org.freedesktop.network1 path=/org/freedesktop/network1 interface=org.freedesktop.DBus.Properties m
ember=Get cookie=2 reply_cookie=0 signature=ss error-name=n/a error-message=n/a
Got message type=method_return sender=:1.2 destination=:1.38 path=n/a interface=n/a member=n/a cookie=52 reply_cookie=2 signature=v error-name=n/a error
-message=n/a
systemd-networkd.service not running in a network namespace (?), skipping netns check.
Bus n/a: changing state RUNNING â CLOSED
Bus n/a: changing state UNSET â OPENING
sd-bus: starting bus by connecting to /run/dbus/system_bus_socket...
Bus n/a: changing state OPENING â AUTHENTICATING
Bus n/a: changing state AUTHENTICATING â HELLO
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=Hello cookie=
1 reply_cookie=0 signature=n/a error-name=n/a error-message=n/a
Got message type=method_return sender=org.freedesktop.DBus destination=:1.39 path=n/a interface=n/a member=n/a cookie=1 reply_cookie=1 signature=s error
-name=n/a error-message=n/a
Bus n/a: changing state HELLO â RUNNING
Sent message type=method_call sender=n/a destination=org.freedesktop.network1 path=/org/freedesktop/network1 interface=org.freedesktop.network1.Manager 
member=Describe cookie=2 reply_cookie=0 signature=n/a error-name=n/a error-message=n/a
Got message type=error sender=:1.2 destination=:1.39 path=n/a interface=n/a member=n/a cookie=53 reply_cookie=2 signature=s error-name=org.freedesktop.D
Bus.Error.InvalidArgs error-message=Invalid argument
Failed to get description: Invalid argument
Bus n/a: changing state RUNNING â CLOSED
#
```

**Comment by yuwata** [comment]
Sorry for late response.

Please provide all relevant .network files and the result of `networkctl status --all -n 0`.

**Comment by dave4445** [comment]
```
$ grep -v '^#' /etc/systemd/networkd.conf 

[Network]

[DHCPv4]

[DHCPv6]
```
```
$ find / -name \*.network |while read f; do echo; echo ==============; echo \# $f:; grep -v '^#' $f; done

==============
# /etc/systemd/network/eth0.network:
[Match]
Name=eth0
[Network]
DHCP=ipv4

==============
# /usr/lib/systemd/network/80-6rd-tunnel.network:


[Match]
Name=6rd-*
Type=sit

[Network]
DHCPPrefixDelegation=yes

==============
# /usr/lib/systemd/network/80-container-host0.network:


[Match]
Virtualization=container
Name=host0

[Network]
DHCP=yes
LinkLocalAddressing=yes
LLDP=yes
EmitLLDP=customer-bridge

[DHCP]
UseTimezone=yes

==============
# /usr/lib/systemd/network/80-container-ve.network:


[Match]
Name=ve-*
Driver=veth

[Network]
Address=0.0.0.0/28
LinkLocalAddressing=yes
DHCPServer=yes
IPMasquerade=both
LLDP=yes
EmitLLDP=customer-bridge
IPv6SendRA=yes

==============
# /usr/lib/systemd/network/80-container-vz.network:


[Match]
Name=vz-*
Driver=bridge

[Network]
Address=0.0.0.0/24
LinkLocalAddressing=yes
DHCPServer=yes
IPMasquerade=both
LLDP=yes
EmitLLDP=customer-bridge
IPv6SendRA=yes

==============
# /usr/lib/systemd/network/80-vm-vt.network:


[Match]
Name=vt-*
Driver=tun

[Network]
Address=0.0.0.0/28
LinkLocalAddressing=yes
DHCPServer=yes
IPMasquerade=both
LLDP=yes
EmitLLDP=customer-bridge
IPv6SendRA=yes

==============
# /usr/lib/systemd/network/80-wifi-adhoc.network:

[Match]
Type=wlan
WLANInterfaceType=ad-hoc

[Network]
LinkLocalAddressing=yes
```
```
$ networkctl status --all -n 0 |cat
- 1: lo                         
                     Link File: n/a
                  Network File: n/a
                          Type: loopback
                         State: carrier (unmanaged)
                  Online state: unknown
              Hardware Address: 00:00:00:00:00:00
                           MTU: 65536
                         QDisc: noqueue
  IPv6 Address Generation Mode: eui64
          Queue Length (Tx/Rx): 1/1
                       Address: 127.0.0.1
                                ::1
             Activation Policy: up
           Required For Online: yes

- 2: eth0                       
                     Link File: /usr/lib/systemd/network/99-default.link
                  Network File: /etc/systemd/network/eth0.network
                          Type: ether
                         State: routable (configured)
                  Online state: online
                          Path: platform-ff4e0000.ethernet
              Hardware Address: f2:33:93:b2:32:61
                           MTU: 1500
                         QDisc: pfifo_fast
  IPv6 Address Generation Mode: eui64
          Queue Length (Tx/Rx): 1/1
              Auto negotiation: yes
                         Speed: 100Mbps
                        Duplex: full
                          Port: mii
                       Address: 192.xxx.xxx.179 (DHCP4 via 192.xxx.xxx.1)
                                2620:xxxx:xxxx:1061::8:b9
                                2620:xxxx:xxxx:1061:f033:93ff:feb2:3261
                                fe80::f033:93ff:feb2:3261
                       Gateway: 192.xxx.xxx.161 (Cisco Systems, Inc)
                                fe80::1061 (Cisco Systems, Inc)
                           DNS: 192.xxx.xxx.1
                                192.xxx.xxx.146
                                2620:xxxx:xxxx::1
                                2620:xxxx:xxxx:1060::10
             Activation Policy: up
           Required For Online: yes
               DHCP4 Client ID: IAID:0x8821af4d/DUID
             DHCP6 Client IAID: 0x8821af4d
             DHCP6 Client DUID: DUID-EN/Vendor:0000ab1164cfb18ec76d56b50000
                  Connected To: office2-sw1 on port 5 (Port #5)
$ 
```

**Comment by dave4445** [comment]
Found the issue... Traced it to `manager_get_route_table_to_string` returning `-EINVAL` for table number 0 which is a valid number, but this function didn't like that.
This is the offending route which originates from the kernel as a v6 unreachable default:
```
$ ip -6 route list default
default via fe80::1061 dev eth0  metric 1024 
unreachable default dev lo  metric -1  error -101
unreachable default dev lo  metric -1  error -101
$  
```
```
        "Routes" : [
                {
                        "Family" : 10,
                        "Destination" : [
                                0,
                                0,
                                0,
                                0,
                                0,
                                0,
                                0,
                                0,
                                0,
                                0,
                                0,
                                0,
                                0,
                                0,
                                0,
                                0
                        ],
                        "DestinationPrefixLength" : 0,
                        "Scope" : 0,
                        "ScopeString" : "global",
                        "Protocol" : 2,
                        "ProtocolString" : "kernel",
                        "Type" : 7,
                        "TypeString" : "unreachable",
                        "Priority" : 4294967295,
                        "Table" : 0,
                        "TableString" : "0",
                        "Preference" : 0,
                        "Flags" : 0,
                        "FlagsString" : "",
                        "ConfigSource" : "foreign",
                        "ConfigState" : "configured"
                }
        ],
```
Patch attached
[002_zero_is_a_valid_route_table.patch.txt](https://github.com/systemd/systemd/files/9872085/002_zero_is_a_valid_route_table.patch.txt)

**Comment by yuwata** [comment]
Thank you for the investigation. The patch looks good to me. Could you submit the change as a PR?

When you submit the change, please add brief comments on `manager_get_route_table_to_string()/_from_string()`, why `_from_string()` refuses 0, but `_to_string()` accepts 0?

**Comment by yuwata** [comment]
Fix is waiting in #25164.
